api/services_handlers_test.go

```diff
@@ -42,40 +43,301 @@ func TestServicesCreateHandler(t *testing.T) {
+			wantStatus: http.StatusBadRequest,
+		},
+		{
+			name:       "InvalidJSON",
+			json:       invalidServiceJSON,
+			wantStatus: http.StatusUnprocessableEntity,
+		},
+		{
+			name:       "ValidJSON",
+			json:       validServiceJSON,
+			wantStatus: http.StatusCreated,
+		},
+	}
+
 	app := helpers.GetApp(t)
-	bts, err := json.Marshal(service)
+
+	for _, testCase := range testCases {
+		t.Run(testCase.name, func(t *testing.T) {
+			req, _ := http.NewRequest("POST", "/services", bytes.NewBuffer(testCase.json))
+			req.Header.Set("Authorization", fmt.Sprintf("KeyPair %v:%v", rootSA.KeyID, rootSA.KeySecret))
+
+			resp := helpers.DoRequest(t, req, app.GetRouter())
+			if resp.Code != testCase.wantStatus {
+				t.Fatalf("Code = %v, want %v", resp.Code, testCase.wantStatus)
+			}
+		})
+	}
+}
+
+func TestServicesCreateHandler_servicePersisted(t *testing.T) {
+	beforeEachServices(t)
+
+	rootSA := helpers.CreateRootServiceAccountWithKeyPair(t, "rootSAKeyPair", "rootSAKeyPair@test.com")
+	saUC := helpers.GetServiceAccountsUseCase(t)
+	sa := &models.ServiceAccount{
+		Name:  "any",
+		Email: "any@email.com",
+	}
+	if err := saUC.Create(sa); err != nil {
+		t.Fatalf("Create() returned error = %v", err)
+	}
+
+	service := &models.Service{
+		Name:                    "Some Service",
+		PermissionName:          "SomeService",
+		CreatorServiceAccountID: sa.ID,
+		AMURL:                   "http://localhost:3333/am",
+	}
+
+	reqJSON, err := json.Marshal(service)
 	if err != nil {
-		t.Errorf("Unexpected error: %s", err.Error())
-		return
-	}
-	req, _ := http.NewRequest("POST", "/services", bytes.NewBuffer(bts))
-	req.Header.Set("Authorization", fmt.Sprintf(
-		"KeyPair %s:%s", rootSA.KeyID, rootSA.KeySecret,
-	))
-	rec := helpers.DoRequest(t, req, app.GetRouter())
-	if rec.Code != http.StatusCreated {
-		t.Errorf("Expected 201. Got %d", rec.Code)
-		return
-	}
-	ssUC := helpers.GetServicesUseCase(t)
-	ss, err := ssUC.List()
+		t.Fatalf("Marshal() returned error = %v", err)
+	}
+
+	req, _ := http.NewRequest("POST", "/services", bytes.NewBuffer(reqJSON))
+	req.Header.Set("Authorization", fmt.Sprintf("KeyPair %s:%s", rootSA.KeyID, rootSA.KeySecret))
+
+	app := helpers.GetApp(t)
+	resp := helpers.DoRequest(t, req, app.GetRouter())
+	if resp.Code != http.StatusCreated {
+		t.Fatalf("Code = %v, want %v", resp.Code, http.StatusCreated)
+	}
+
+	servicesUC := helpers.GetServicesUseCase(t)
+	services, err := servicesUC.List()
 	if err != nil {
-		t.Errorf("Unexpected error: %s", err.Error())
-		return
-	}
-	if len(ss) != 1 {
-		t.Errorf("Expected to have 1 service. Got %d", len(ss))
-		return
-	}
-	if ss[0].Name != "Some Service" {
-		t.Errorf("Expected service name to be Some Service. Got %s", ss[0].Name)
-		return
-	}
-	if ss[0].PermissionName != "SomeService" {
-		t.Errorf(
-			"Expected service permission name to be SomeService. Got %s",
-			ss[0].PermissionName,
-		)
-		return
+		t.Fatalf("List() returned error = %v", err)
+	}
+
+	got := services[0]
```

Comment:
This is a lot better, but I would diff the entire list, not only the first index.

Eg:

```go
gotService := services[0]
wantService := service
wantService.ID = gotService.ID
// ...
want := []models.Service{wantService}
if diff  := pretty.Compare(services, want); diff != "" {
  // ...
}
```